File: Examples/CSharp/MongoDB/CRUD/Client/MongoDB_Client_Builder.cs
using MongoDB.Driver;

namespace MongoDB.CRUD.Client;

public class MongoDB_Client_Builder : IMongoDB_Client_Builder
{
    public MongoDB_Client_Builder(MongoClientSettings mongoClientSettings)
    {
        MongoClientSettings = mongoClientSettings;
    }

    private MongoClientSettings MongoClientSettings { get; }

    public IMongoClient Build()
    {
        return new MongoClient(MongoClientSettings);
    }
}

File: Examples/Java/Containers/MongoDB-REST/src/main/java/com/api/mongodb/rest/service/mongodb/crud/MongoDB_Client_Builder.java
package com.api.mongodb.rest.service.mongodb.crud;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;

public class MongoDB_Client_Builder implements IMongoDB_Client_Builder {
    private final MongoClientSettings mongoClientSettings;

    public MongoDB_Client_Builder(String connectionString) {
        this.mongoClientSettings = MongoClientSettings.builder()
            .applyConnectionString(new ConnectionString(connectionString))
            .build();
    }

    public MongoDB_Client_Builder(MongoClientSettings mongoClientSettings) {
        this.mongoClientSettings = mongoClientSettings;
    }

    public MongoClient build() {
        System.out.println("MongoDB Client Settings: " + mongoClientSettings);
        return MongoClients.create(mongoClientSettings);
    }
}
